Disable breaking if no haptic feedback support which disallows gamecontroller API

Index: src/SDL.c
--- src/SDL.c.orig
+++ src/SDL.c
@@ -279,9 +279,6 @@ SDL_InitSubSystem(Uint32 flags)
         }
         SDL_PrivateSubsystemRefCountIncr(SDL_INIT_GAMECONTROLLER);
         flags_initialized |= SDL_INIT_GAMECONTROLLER;
-#else
-        SDL_SetError("SDL not built with joystick support");
-        goto quit_and_error;
 #endif
     }
 
